src/mmw/apps/modeling/tasks.py

```diff
@@ -146,8 +146,19 @@ def analyze_pointsource(area_of_interest):
     return [point_source_pollution(area_of_interest)]
 
 
+def ensure_aoi_4d(area_of_interest):
+    # In rare instances, this method receives a five-dimensional array for these
+    # coordinates. This check is here to verify whether the `area_of_interest` is
+    # four- or five-dimensional, then assigns `pairs` appropriately.
+    if type(area_of_interest['coordinates'][0][0][0][0]) is list:
```

Comment:
Actually it looks like some of the AOIs which get sent from the frontend are declared to be type `MultiPolygon'`, too, but send in 4d arrays. This one:

![screen shot 2016-09-16 at 9 19 28 am](https://cloud.githubusercontent.com/assets/4165523/18587190/b2bd958e-7bee-11e6-81f5-294c7abc8345.png)

Sends this JSON blob to the API:

```json
{"type":"MultiPolygon","coordinates":[[[[-75.21211271624472,40.003202106163144],[-75.20037526745891,40.003202106163144],[-75.20037526745891,40.01219249993579],[-75.21211271624472,40.01219249993579],[-75.21211271624472,40.003202106163144]]]]}
```

... and the boundary geojson for Philadelphia county sends this:

```json
{"type":"MultiPolygon","coordinates":[[[[-75.280266,39.974964],[-75.276482,39.976959],[-75.275871,39.977245],[-75.273552,39.978279],[-75.271309,39.979275],[-75.269257,39.980186],[-75.267973,39.980757],[-75.265511,39.981857],[-75.259476,39.984955],[-75.255676,39.987499],[-75.253999,39.988304],[-75.251438,39.989504],[-75.25,39.99018],[-75.249214,39.990547],[-75.247948,39.991138],[-75.237331,39.996128],[-75.235298,39.99707],[-75.230896,39.99913],[-75.227462,40.000746],[-75.226402,40.001255],[-75.220665,40.003944],[-75.213745,40.00716],[-75.207993,40.009842],[-75.207275,40.010189],[-75.206968,40.011372],[-75.207599,40.012719],[-75.208008,40.012997],[-75.212219,40.015461],[-75.214752,40.017784],[-75.218056,40.020943],[-75.219583,40.021713],[-75.222153,40.022614],[-75.223694,40.023712],[-75.226652,40.025267],[-75.228149,40.026031],[-75.232628,40.028355],[-75.234604,40.029926],[-75.237762,40.031934],[-75.238068,40.032127],[-75.241493,40.035141],[-75.245598,40.037655],[-75.249039,40.041848],[-75.251298,40.043505],[-75.254678,40.045924],[-75.255098,40.046122],[-75.257655,40.047611],[-75.264431,40.054094],[-75.255006,40.062704],[-75.243288,40.073494],[-75.232952,40.083618],[-75.2305,40.086022],[-75.22361,40.092907],[-75.213165,40.086872],[-75.21273,40.086388],[-75.211868,40.086086],[-75.209396,40.084927],[-75.200378,40.07959],[-75.197601,40.077991],[-75.189117,40.073124],[-75.188461,40.07275],[-75.186096,40.075123],[-75.182053,40.079102],[-75.180901,40.080219],[-75.179756,40.081333],[-75.177696,40.083378],[-75.176438,40.084649],[-75.172951,40.082581],[-75.169914,40.080818],[-75.163399,40.07695],[-75.161285,40.075764],[-75.158417,40.074169],[-75.154076,40.071732],[-75.151596,40.070328],[-75.148331,40.068485],[-75.146683,40.067547],[-75.145187,40.066692],[-75.143117,40.065493],[-75.140709,40.064133],[-75.139427,40.063358],[-75.136719,40.061821],[-75.135124,40.060989],[-75.130135,40.057796],[-75.129021,40.057091],[-75.12632,40.055481],[-75.124916,40.054684],[-75.122147,40.053177],[-75.118652,40.051186],[-75.117104,40.050304],[-75.112363,40.047563],[-75.110985,40.046769],[-75.109431,40.045874],[-75.109406,40.04584],[-75.10789,40.048195],[-75.10323,40.051355],[-75.102413,40.051792],[-75.100935,40.052699],[-75.100896,40.052723],[-75.095078,40.056091],[-75.093483,40.057259],[-75.087425,40.063599],[-75.089378,40.064705],[-75.09037,40.065285],[-75.096733,40.068989],[-75.090317,40.075115],[-75.086906,40.078648],[-75.082456,40.082884],[-75.077805,40.08577],[-75.076363,40.087265],[-75.07127,40.092532],[-75.069456,40.094411],[-75.058342,40.105892],[-75.054321,40.108692],[-75.0493,40.112199],[-75.048542,40.112729],[-75.043856,40.115997],[-75.037842,40.120213],[-75.024891,40.129216],[-75.020571,40.132274],[-75.015066,40.137992],[-75.009117,40.134355],[-75.003255,40.131012],[-74.997885,40.127816],[-74.994187,40.131516],[-74.993595,40.132266],[-74.992288,40.130682],[-74.993687,40.126584],[-74.989014,40.12607],[-74.987193,40.123286],[-74.983523,40.123671],[-74.982213,40.122057],[-74.982748,40.120887],[-74.980058,40.119883],[-74.973219,40.119891],[-74.970119,40.117201],[-74.964526,40.118372],[-74.96408,40.117766],[-74.963129,40.114978],[-74.96345,40.110013],[-74.965194,40.107777],[-74.963491,40.10674],[-74.961156,40.103446],[-74.960807,40.101572],[-74.96163,40.097804],[-74.961049,40.097431],[-74.958749,40.096929],[-74.956063,40.09557],[-74.955777,40.095103],[-74.956506,40.093115],[-74.958036,40.089665],[-74.958392,40.084508],[-74.958284,40.084145],[-74.958841,40.082503],[-74.962737,40.077839],[-74.964883,40.077273],[-74.969879,40.076657],[-74.970537,40.076281],[-74.971028,40.075981],[-74.971268,40.073744],[-74.977152,40.068223],[-74.980525,40.065702],[-74.980688,40.065007],[-74.980718,40.06472],[-74.980378,40.060349],[-74.984302,40.05585],[-74.983904,40.054994],[-74.982202,40.053804],[-74.979695,40.053399],[-74.97431985326,40.04889913332],[-74.974713,40.048711],[-74.983540729443,40.0429537851459],[-74.983913,40.042711],[-74.9869972233608,40.0393469567366],[-74.988884,40.037289],[-74.990215350337,40.0363725639031],[-75.007469,40.024496],[-75.011115,40.021311],[-75.0137957681416,40.0202143221239],[-75.0138034952734,40.0202111610245],[-75.015515,40.019511],[-75.0211950666745,40.018207388098595],[-75.0304678308145,40.0160792289794],[-75.034216,40.015219],[-75.042889,40.012032],[-75.04296644952109,40.0119934503812],[-75.043218536113,40.01186797714539],[-75.04709,40.009941],[-75.04712071178679,40.00991486944879],[-75.053405,40.004568],[-75.05372388596429,40.004080276035694],[-75.058089,39.997404],[-75.059017,39.992512],[-75.0597922716131,39.9915372740575],[-75.064175,39.986027],[-75.0646459796169,39.9856415234906],[-75.0676640966386,39.9831713248389],[-75.0686043012039,39.982401807940505],[-75.068802,39.98224],[-75.075605,39.978041],[-75.085831,39.975102],[-75.093718,39.974412],[-75.09371907941859,39.9744116926869],[-75.1075409360153,39.970476576233395],[-75.108119,39.970312],[-75.11277110412401,39.968258553445004],[-75.1189552837088,39.9655288462145],[-75.11922,39.965412],[-75.1238629772435,39.96281916855229],[-75.1254163616611,39.9619516941373],[-75.12692,39.961112],[-75.13012,39.958712],[-75.1302170895175,39.9585892103161],[-75.13352,39.954412],[-75.13397386650959,39.95290598840011],[-75.13572,39.947112],[-75.13612,39.933912],[-75.13588759459459,39.932517567567594],[-75.13534618918919,39.9292691351351],[-75.135022972973,39.9273298378378],[-75.13502,39.927312],[-75.13282,39.921612],[-75.13012,39.917013],[-75.12792,39.911813],[-75.1290044117647,39.9074753529412],[-75.13082,39.900213],[-75.13342,39.896213],[-75.1359848633979,39.893195957332196],[-75.1400064847137,39.888465333817194],[-75.140221,39.888213],[-75.142421,39.886413],[-75.14532392051939,39.8842841916191],[-75.145421,39.884213],[-75.150721,39.882713],[-75.17453546617179,39.8821969289728],[-75.183023,39.882013],[-75.18926826703719,39.8807242941034],[-75.189323,39.880713],[-75.195018,39.877521],[-75.1952172947686,39.877343945193196],[-75.199857,39.873222],[-75.204087,39.871204],[-75.21151593114269,39.8670515524317],[-75.216039,39.871115],[-75.215787,39.875052],[-75.215722,39.876087],[-75.21977,39.875479],[-75.223763,39.875858],[-75.229985,39.876847],[-75.234122,39.877181],[-75.23637,39.876187],[-75.24047,39.872381],[-75.241746,39.872585],[-75.24682,39.875142],[-75.248431,39.876485],[-75.253515,39.876015],[-75.261245,39.876484],[-75.262539,39.876578],[-75.263328,39.879712],[-75.264928,39.881112],[-75.264433,39.882167],[-75.261937,39.883],[-75.264602,39.884737],[-75.262844,39.885682],[-75.265153,39.887142],[-75.264878,39.887927],[-75.262485,39.888406],[-75.261743,39.891774],[-75.259112,39.892436],[-75.256789,39.895189],[-75.257362,39.895844],[-75.255093,39.898756],[-75.255597,39.899596],[-75.251288,39.904403],[-75.253492,39.905585],[-75.2525,39.906614],[-75.250319,39.907626],[-75.249929,39.909352],[-75.251498,39.910051],[-75.2472,39.913578],[-75.24694,39.916871],[-75.247341,39.918541],[-75.247484,39.919242],[-75.247459,39.921343],[-75.247759,39.921708],[-75.247321,39.922178],[-75.245928,39.924474],[-75.244336,39.92625],[-75.243643,39.930897],[-75.243369,39.931908],[-75.241909,39.933544],[-75.237617,39.932526],[-75.236833,39.933768],[-75.234979,39.936222],[-75.235002,39.938127],[-75.24021,39.938283],[-75.239411,39.940608],[-75.240367,39.942171],[-75.243026,39.942211],[-75.243542,39.942306],[-75.246822,39.94297],[-75.247028,39.943414],[-75.247208,39.943904],[-75.250118,39.94697],[-75.251609,39.951466],[-75.251577,39.952127],[-75.250816,39.953475],[-75.249006,39.956907],[-75.249499,39.957189],[-75.250883,39.959024],[-75.248546,39.960269],[-75.247688,39.962817],[-75.248056,39.964074],[-75.250313,39.965069],[-75.25414,39.965404],[-75.25711,39.963719],[-75.259078,39.965692],[-75.260491,39.966313],[-75.265448,39.965257],[-75.267386,39.966215],[-75.270002,39.967479],[-75.270598,39.96866],[-75.275394,39.969575],[-75.27754,39.971127],[-75.280266,39.974964]]]]}
```